airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java

```diff
@@ -232,6 +238,38 @@ public WorkspaceRetentionConfig getWorkspaceRetentionConfig() {
     return new WorkspaceRetentionConfig(minDays, maxDays, maxSizeMb);
   }
 
+  @Override
+  public List<WorkerPodToleration> getWorkerPodTolerations() {
+    // Format:
+    // Tolerations separated by ;;
+    // Each toleration has k=v format for necessary fields separated by ,
+    // t1-k1=v1,t1-k2=v2,t1-k3=v3;;t2-k1=v1,t2-k2=v2,t2-k3=v3 ....
+    // we need at-least key, effect and operator to be present. value is optional
+    // we do not
+    String tolerations = getEnvOrDefault(WORKER_POD_TOLERATIONS, "");
+    return Strings.isNullOrEmpty(tolerations) ? List.of() : Splitter.on(";;")
+        .splitToStream(tolerations)
+        .filter(tolerationStr -> !Strings.isNullOrEmpty(tolerationStr))
+        .map(tolerationStr -> {
+          Map<String, String> tolerationMap = Splitter.on(",")
+              .splitToStream(tolerationStr)
+              .map(s -> s.split("="))
+              .collect(Collectors.toMap(s -> s[0], s -> s[1]));
+          if (tolerationMap.containsKey("key") && tolerationMap.containsKey("effect") && tolerationMap.containsKey("operator")) {
+            return new WorkerPodToleration(tolerationMap.get("key"),
+                tolerationMap.get("effect"),
+                tolerationMap.get("value"),
+                tolerationMap.get("operator"));
+          } else {
+            LOGGER.warn("Ignoring toleration {}, missing one of key,effect or operator",
+                tolerationStr);
+            return null;
+          }
+        })
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
```

Comment:
This block is tough to read.

What about splitting into different statements? One natural split is 1) parse tolerations from env var 2) map tolerations into `WorkerPodToleration` objects.

The map function can also be pulled out into a separate function to keep things simple. What do you think?

Reply:
Yeah sounds good, will do.